alignak/objects/schedulingitem.py

```diff
@@ -1970,3 +1972,275 @@ def fill_data_brok_from(self, data, brok_type):
+            self.problem_has_been_acknowledged = True
+            if sticky == 2:
+                sticky = True
+            else:
+                sticky = False
+            ack = Acknowledge(self, sticky, notify, persistent, author, comment, end_time=end_time)
+            self.acknowledgement = ack
+            if self.my_type == 'host':
+                comment_type = 1
+            else:
+                comment_type = 2
+            comm = Comment(self, persistent, author, comment,
+                           comment_type, 4, 0, False, 0)
+            self.add_comment(comm)
+            self.broks.append(self.get_update_status_brok())
+
+    def check_for_expire_acknowledge(self):
+        """
+        If have acknowledge and is expired, delete it
+
+        :return: None
+        """
+        if (self.acknowledgement and
+                self.acknowledgement.end_time != 0 and
+                self.acknowledgement.end_time < time.time()):
+            self.unacknowledge_problem()
+
+    def unacknowledge_problem(self):
+        """
+        Remove the acknowledge, reset the flag. The comment is deleted except if the acknowledge
+        is defined to be persistent
+
+        :return: None
+        """
+        if self.problem_has_been_acknowledged:
+            logger.debug("[item::%s] deleting acknowledge of %s",
+                         self.get_name(),
+                         self.get_full_name())
+            self.problem_has_been_acknowledged = False
+            # Should not be deleted, a None is Good
+            self.acknowledgement = None
+            # del self.acknowledgement
+            # find comments of non-persistent ack-comments and delete them too
+            for comm in self.comments:
+                if comm.entry_type == 4 and not comm.persistent:
+                    self.del_comment(comm._id)
+            self.broks.append(self.get_update_status_brok())
+
+    def unacknowledge_problem_if_not_sticky(self):
+        """
+        Remove the acknowledge if it is not sticky
+
+        :return: None
+        """
+        if hasattr(self, 'acknowledgement') and self.acknowledgement is not None:
+            if not self.acknowledgement.sticky:
+                self.unacknowledge_problem()
+
+    def raise_alert_log_entry(self):
+        """Raise ALERT entry (critical level)
+        It's defined in right objects (Host and Service)
+
+        :return: None
+        """
+        pass
+
+    def is_state(self, status):
+        """Return if status match the current item status
+
+        :param status: status to compare. Usually comes from config files
+        :type status: str
+        :return: True
+        :rtype: bool
+        """
+        return True
+
+    def raise_freshness_log_entry(self, t_stale_by, t_threshold):
+        """Raise freshness alert entry (warning level)
+        Format is : "The results of host '*get_name()*' are stale by *t_stale_by*
```

Comment:
I think you have to fix some docstring to replace host by item and say it's a abstract method or something.